src/WebJobs.Script/ManagedDependencies/ManagedDependencyProvider.cs

```diff
@@ -0,0 +1,153 @@
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.ManagedDependencies
+{
+    internal class ManagedDependencyProvider : IManagedDependencyProvider
+    {
+        private const string ManagedDependenciesPropertiesName = "managedDependencies";
+
+        /// <summary>
+        /// Returns the managed dependecies meta data
+        /// </summary>
+        /// <param name="hostFilePath">The configuration path for the managed dependencies</param>
+        /// <param name="environment">The environment object reprenested as inteface <see cref="IEnvironment"/> </param>
+        /// <param name="managedDependencyOptions">Configurations options for managed dependencies</param>
+        /// <returns>The IEnumerable list of <see cref="Dependency"/></returns>
+        public IEnumerable<Dependency> GetDependencies(string hostFilePath,
+            IEnvironment environment,
+            IOptions<ManagedDependencyOptions> managedDependencyOptions)
+        {
+            if (string.IsNullOrWhiteSpace(hostFilePath)
+                || !File.Exists(hostFilePath)
+                || managedDependencyOptions == null)
+            {
+                return Enumerable.Empty<Dependency>();
+            }
+
+            DependentLibraries managedDependenciesFromHostFile = DeserializeJsonFile<DependentLibraries>(hostFilePath);
+            if (managedDependenciesFromHostFile.ManagedDependencies == null
+                || !managedDependenciesFromHostFile.ManagedDependencies.Any())
+            {
+                return Enumerable.Empty<Dependency>();
+            }
+
+            var dependencyInstallationList = new List<Dependency>();
+            foreach (var dependentLibrary in managedDependenciesFromHostFile.ManagedDependencies)
+            {
+                var dependency = new Dependency();
+                var moduleName = dependentLibrary.Name;
+                var version = dependentLibrary.Version;
+
+                dependency.Name = moduleName;
+                dependency.Version = version;
+
+                var isDownloadRequired = false;
+                var dependencyDownloadPath = default(string);
+
+                //If platform is windows and app service environment
+                //and if the Az modules have been installed through rapid update
+                if (environment.IsAppServiceWindowsEnvironment()
+                    && Directory.Exists(managedDependencyOptions.Value.RapidUpdateAzModulePath))
+                {
+                    var rapidUpdateAzModuleVersionPath = Path.Combine(managedDependencyOptions.Value.RapidUpdateAzModulePath, ScriptConstants.InstalledDependencyInfoFileName);
+                    var installedModule = DeserializeJsonFile<InstalledDependency>(rapidUpdateAzModuleVersionPath);
+                    if (!string.Equals(installedModule.InstalledVersion.Split('.')[0], version, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
The logic to figure out if a installed module needs to be part of the language worker. Figuring out version for a module is different for every language. 
Host  responsibility is to
- Parse  host.json
- Send info on managed assembly path depending on the platform app is running
- Send version information
Language Worker responsibility:
- Use the information sent by the host to figure out if modules need to downloaded or not

Apologies, I missed this in the initial review.

Reply:
Figuring out installed modules on the host also requires host and a language worker to be running on the same machine. In future, it should be possible to run host and language workers on different machine as grpc already supports it.